middlewares/empty_backend_handler_test.go

```diff
@@ -0,0 +1,70 @@
+package middlewares
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/containous/traefik/testhelpers"
+	"github.com/vulcand/oxy/roundrobin"
+)
+
+func TestEmptyBackendHandler(t *testing.T) {
+	tests := []struct {
+		amountServer   int
+		wantStatusCode int
+	}{
+		{
+			amountServer:   0,
+			wantStatusCode: http.StatusServiceUnavailable,
+		},
+		{
+			amountServer:   1,
+			wantStatusCode: http.StatusOK,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(fmt.Sprintf("amount servers %d", test.amountServer), func(t *testing.T) {
+			t.Parallel()
+
+			nextHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusOK)
+			})
+			handler := NewEmptyBackendHandler(&healthCheckLoadBalancer{test.amountServer}, nextHandler)
+
+			recorder := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "http://localhost", nil)
```

Comment:
Always. So thanks for reminding me constantly Timo. At some point I will also learn it, don't give up the hope :D 🐼